components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ResendCodeApiServiceImpl.java

```diff
@@ -64,7 +64,7 @@ public Response resendCodePost(ResendCodeRequestDTO resendCodeRequestDTO) {
         }
 
         if (notificationResponseBean == null || StringUtils.isBlank(notificationResponseBean.getKey())) {
```

Comment:
here we need to check the value for the key in a separate if block.

Reply:
There is an uncertainity on the error codes returned checked in the current test case. Will have a revisit on the test case scenarioes and get back to this.